feat: Ability to add validators directly using deposit function

PhilWindle opened the pull request:
This PR adds an additional cli command for adding a validator to the rollup by calling `deposit` directly.

Fix TMNT-292

aminsammara commented:
I talked to Sean yesterday, there is 0 sense to why the standard `add-l1-validator` is calling the staking asset handler since that requires a zkPassport proof which you cannot pass via the cli. My recommendation is to keep the name `add-l1-validator` and replace the contents with what you have here @PhilWindle

alexghr commented on yarn-project/cli/public_include_metric_prefixes.json:
This path needs to be updated here https://github.com/AztecProtocol/aztec-packages/blob/next/spartan/terraform/deploy-telemetry/main.tf#L93

alexghr commented on yarn-project/cli/src/cmds/l1/index.ts:
```diff
@@ -326,6 +327,46 @@ export function injectCommands(program: Command, log: LogFn, debugLogger: Logger
+      'The BN254 scalar field element used as a secret key for BLS signatures. Will be associated with the attester address.',
+      parseBigint,
+    )
+    .option('--moveWithLatestRollup', 'Whether to move with the latest rollup', true)
```
```suggestion
    .option('--move-with-latest-rollup', 'Whether to move with the latest rollup', true)
```

PhilWindle commented:
> I talked to Sean yesterday, there is 0 sense to why the standard `add-l1-validator` is calling the staking asset handler since that requires a zkPassport proof which you cannot pass via the cli. My recommendation is to keep the name `add-l1-validator` and replace the contents with what you have here @PhilWindle

Ok, will do

alexghr commented on yarn-project/cli/src/cmds/l1/index.ts:
```diff
@@ -326,6 +327,46 @@ export function injectCommands(program: Command, log: LogFn, debugLogger: Logger
+      parseBigint,
+    )
+    .option('--moveWithLatestRollup', 'Whether to move with the latest rollup', true)
+    .option('--rollup <string>', 'Rollup contract address or network name')
```
I propose making this only take an optional rollup address. The `--network` flag/`NETWORK` env var is already handled at the top of the aztec binary. We can just check if we have a `REGISTRY_CONTRACT_ADDRESS` env var (via enrichment) and take the canonical rollup from it.

PhilWindle replied:
Yes, good idea.

alexghr reviewed (changes requested):
Looks good but the terraform in `deploy-telemetry` needs to be updated with the new file path.

The other comments are minor

PhilWindle committed:
feat: Ability to add validators directly using deposit function

This PR adds an additional cli command for adding a validator to the rollup by calling `deposit` directly.

Fix TMNT-292

Co-authored-by: Alex Gherghisan <alexghr@users.noreply.github.com>
Co-authored-by: Mitchell Tracy <mitchellftracy@gmail.com>
